lib/prometheus_server.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+const Prometheus = require('prom-client');
+const AggregatorRegistry = new Prometheus.AggregatorRegistry();
+const HTTP = require('http');
+
+class MetricsEndpoint {
+    constructor(config) {
+        this.server = HTTP.createServer((req, res) => {
+            AggregatorRegistry.clusterMetrics((err, metrics) => {
```

Comment:
This will be needed only if we're running in the cluster mode configuration? see previous comment.